File: src/EduTwin.DAL/CurriculumAndQuestions/Curriculum.cs
using System;
using EduTwin.Contracts.CurriculumAndQuestions;
using EduTwin.DAL.Persistence.Models;
using EduTwin.DAL.Organization;

namespace EduTwin.DAL.CurriculumAndQuestions;

public class Curriculum : IMutableTenantAggregate
{
    public Guid CurriculumId { get; set; }
    public Guid CenterId { get; set; }
    public Guid TeacherId { get; set; }
    public Guid SubjectId { get; set; }
    public string Title { get; set; } = null!;
    public string? Description { get; set; }
    public string? SourceFile { get; set; }
    public ReviewStatus ReviewStatus { get; set; }

    // Audit and MTA
    public DateTime CreatedAt { get; set; }
    public Guid? CreatedBy { get; set; }
    public DateTime UpdatedAt { get; set; }
    public Guid? UpdatedBy { get; set; }
    public bool IsDeleted { get; set; }
    public DateTime? DeletedAt { get; set; }
    public Guid? DeletedBy { get; set; }
    public ulong RowVersion { get; set; }

    // Navigations
    public Teacher? Teacher { get; set; }
    public Subject? Subject { get; set; }
}

File: web/edutwin-web/src/types/curriculum.ts
export type ReviewStatus = "Draft" | "Published" | "Archived";

export interface Curriculum {
  curriculumId: string;
  subjectId: string;
  teacherId: string | null;
  title: string;
  description: string | null;
  reviewStatus: ReviewStatus;
  classIds: string[];
  nodeIds: string[];
  rowVersion: string;
}

export interface CreateCurriculumRequest {
  teacherId?: string | null;
  subjectId: string;
  title: string;
  description?: string;
  nodeIds: string[];
}

export interface UpdateCurriculumRequest {
  title: string;
  description?: string;
  rowVersion: string;
}

export interface PublishCurriculumRequest {
  rowVersion: string;
}

export interface UpdateCurriculumClassesRequest {
  classIds: string[];
  rowVersion: string;
}

export interface UpdateCurriculumNodesRequest {
  nodeIds: string[];
  rowVersion: string;
}
